joshua/joshua_agent.py

```diff
@@ -519,6 +523,30 @@ def _run_ensemble(
 
         #    print('{} Running ensemble in dir: {}'.format(threading.current_thread().name, work_dir),file=getFileHandle())
 
+        # Check and set the Pod label when run inside a k8s pod
+        # env['HOSTNAME'] contains the pod name in k8s
+        namespace = ""
+        k8s_namespace_file = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+        if env['HOSTNAME'] and os.path.isfile(k8s_namespace_file):
+            pod = env['HOSTNAME']
+            with open(k8s_namespace_file, 'r') as f:
+                namespace = f.read()
+
+            # Get the cluster config
+            config.load_incluster_config()
+            v1 = client.CoreV1Api()
+
+            # If last_test label is set to true, exit right away
+            pod_desc = v1.read_namespaced_pod(pod, namespace)
+            try:
+                if pod_desc.metadata.labels['last_test']  == "true":
```

Comment:
```suggestion
                if pod_desc.metadata.labels.get('last_test', 'false') == "true":
```

Is there a reason to put this into a try statement instead of just using a default value and checking if the labels are set? Do we actually care about the value or would a check if the key exists be enough?